benchmarking/report.py

```diff
@@ -0,0 +1,173 @@
+
+    Returns
+    ----------
+    tex_num   : str
+        TeX-readable num with thin space
+    
+    """
+    num = '{0:.9f}'.format(num)
+    dec_pos= num.find('.') 
+    if dec_pos != -1:
+        if len(num[dec_pos+4:]) > 0:
+            tex_num = (num[:dec_pos+4] + '\\,' + num[dec_pos+4:dec_pos+7])
+    else:
+        tex_num = num
+    return tex_num
+
+def write_document(results, output, labels=None, match=None):
+    """Write LaTeX document section with preamble, run info, and table entries 
+    for all benchmark data comparing the calculated and experimental values 
+    along with uncertainties.
+
+    Parameters
+    ----------
+    results : iterable of str
+        Name of a results file produced by the benchmarking script.
+    output    : str
+        Name of the file to be written, ideally a .tex file
+    labels: iterable of str
+        Labels for each dataset to use in legend
+    match : str
+        Pattern to match benchmark names to
+
+    Returns
+    -------
+    None
+
+    """
+    # Open and write file
+    if output is None:
+        output = 'report.tex'
+    tex = open(output, 'w')
+
+    # Define document preamble
+    preamble = [
+                '\\documentclass[12pt]{article}', 
+                '\\usepackage[letterpaper, margin=1in]{geometry}',
+                '\\usepackage{dcolumn}',
+                '\\usepackage{tabularx}',
+                '\\usepackage{booktabs}',
+                '\\usepackage{longtable}',
+                '\\usepackage{fancyhdr}',
+                '\\setlength\\LTcapwidth{5.55in}',
+                '\\setlength\\LTleft{0.5in}',
+                '\\setlength\\LTright{0.5in}'
+                ]
+
+    # Define document start and end snippets
+
+    doc_start = ['\\begin{document}', '\\part*{Benchmark Results}']
+    
+    doc_end = ['\\end{document}']
+
+    if labels is None:
+        labels = [Path(f).name for f in results]
+    
+    # Read data from spreadsheets
+    dataframes = {}
+    for csvfile, label in zip(results, labels):
+        dataframes[label] = get_result_dataframe(csvfile).dropna() 
+        #fills dataframe with a key (label) and value (entire results df)
+
+    # Get model keff and uncertainty from ICSBEP
+    icsbep = get_icsbep_dataframe()
+
+    # Determine common benchmarks
+    base = labels[0]
+    index = dataframes[base].index #get raw icsbep case names 
+
+    for df in dataframes.values(): #get the case/value column of the dataframes
+        index = index.intersection(df.index) #gives all of the cases that need to be displayed
+
+    # Applying matching as needed (DEPRECATED)
+    if match is not None:
+        cond = index.map(lambda x: fnmatch(x, match))
+        index = index[cond]
+
+    # Custom Table Caption
+    caption = '\\caption{\\label{tab:1} Criticality (' + labels[0] + ') Benchmark Results}\\\\'
+    # Define Table Entry
+    table = [
+            'Table \\ref{tab:1} uses (nuclear data set info here) to evaluate ICSBEP benchmarks.',
```

Comment:
```suggestion
            'Table \\ref{tab:1} uses (nuclear data set info here) to evaluate ICSBEP benchmarks.',
```
Perhaps a separate topic but if hdf5 nuclear data files had some tags in them so we could get the library name and the version from the hdf5 files automatically.

Reply:
Thanks for taking a look. I'll add the automatic `openmc.__version__` that you suggested. As for the nuclear data files, I was trying to imitate the style of [this LANL report on MCNP](https://mcnp.lanl.gov/pdf_files/TechReport_2022_LANL_LA-UR-22-32951Rev.1_JoseyClarkEtAl.pdf), section 4.2 (image below). I'm not sure if there is a tag in the hdf5 files, but that would be a nice thing to add!

![image](https://github.com/openmc-dev/validation/assets/64677361/c9429369-11c1-4666-a986-58d5bb13af22)